src/engine.rs

```diff
@@ -234,6 +234,7 @@ pub fn run(params: FunctionRunParams) -> Result<FunctionRunResult> {
         output,
         profile: profile_data,
         scale_factor,
+        success: error_logs.is_empty(),
```

Comment:
This works but I wonder if it would be safer to use `module_result.is_err()` instead? This logic relies on a side-effect of the module result being an error.